.github/workflows/debug-builds.yaml

```diff
@@ -0,0 +1,105 @@
+name: Build and package python Weather
+
+on: push
+
+jobs:
+  build-mac:
+    runs-on: macos-latest
+    env:
+      PYTHONVERSION: "3.11.1"
+      MACOSX_DEPLOYMENT_TARGET: "10.9"
+      PYI_BUILD_MODE: "onedir"
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Cache Python download
+      id: cache-python-download
+      uses: actions/cache@v3
+      with:
+        path: ~/python.pkg
+        key: macOS-Python-${{ env.PYTHON_VERSION }}
+
+    - name: Get Python
+      run: |
+       curl https://www.python.org/ftp/python/${PYTHONVERSION}/python-${PYTHONVERSION}-macos11.pkg -o ~/python.pkg
+    - name: Install Python
+      run: sudo installer -pkg ~/python.pkg -target /
+
+    - name: Install Python dependencies and build
+      run: |
+        python3 --version
+        pip3 install --upgrade pip wheel
+        pip3 install -r requirements.txt
+        pip3 install pyinstaller
+        pyinstaller weather.spec
+      
+    - name: Log files
+      run: |
+        ls dist
+        ls dist/Weather
+        mkdir Weather
+        mv ./dist/Weather.app ./Weather/Weather.app
+        ln -s /Applications ./Weather/Applications
+        hdiutil create -srcfolder ./Weather Weather.dmg
+        
+    - name: Upload a Build Artifact
+      uses: actions/upload-artifact@v3.1.2
+      with:
+        # Artifact name
+        name: Weather-Mac
```

Comment:
Now, if anybody is wondering, the artifacts are zipped, always. It doesn't show the extension though. Protocol, I guess. Same for the other two steps for the other two OSes.